server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/dto/DLPConfigurationDTOTest.java

```diff
@@ -0,0 +1,110 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.dto;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.dlp.api.DLPConfigurationItem;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+class DLPConfigurationDTOTest {
+
+    private static final String ID = "id";
+    private static final String EXPRESSION = "expression";
+    private static final String EXPLANATION = "explanation";
+
+    @Test
+    void toDTOsShouldBeSetAllFields() {
+        List<DLPConfigurationDTO> dtos = 
+            DLPConfigurationDTO.toDTOs(
+                ImmutableList.of(
+                    DLPConfigurationItem.builder()
+                    .id(DLPConfigurationItem.Id.of(ID))
+                    .expression(EXPRESSION)
+                    .explanation(EXPLANATION)
+                    .targetsSender(Optional.of(true))
+                    .targetsRecipients(Optional.of(true))
+                    .targetsContent(Optional.of(true))
+                    .build()));
+
+        DLPConfigurationDTO dto = dtos.get(0);
```

Comment:
Maybe that's a simptome that we need a `DLPConfigurationDTO from(DLPConfigurationItem item) method`

(and let the routes convert the lists)

Reply:
We have, but it's private, i will make it public.
I place `DLPConfigurationItem` <-> `DLPConfigurationItemDTO` conversions both single instance and list in static methods of `DLPConfigurationItemDTO`. And I think it is okay for those util methods, can be used by tests and the router.